azurelinuxagent/ga/exthandlers.py

```diff
@@ -1672,15 +1672,25 @@ def collect_ext_status(self, ext):
             data_str, data = self._read_status_file(ext_status_file)
         except ExtensionStatusError as e:
             msg = ""
+            ext_status.status = ValidHandlerStatus.error
+
             if e.code == ExtensionStatusError.CouldNotReadStatusFile:
                 ext_status.code = ExtensionErrorCodes.PluginUnknownFailure
                 msg = u"We couldn't read any status for {0} extension, for the sequence number {1}. It failed due" \
                       u" to {2}".format(self.get_full_name(ext), seq_no, ustr(e))
-            elif ExtensionStatusError.InvalidJsonFile:
+            elif e.code == ExtensionStatusError.InvalidJsonFile:
```

Comment:
what exactly was this code doing before this change? It looks like it's checking the truth-iness of a constant... Is that correct?

Reply:
I think it was just a bug lol. Somehow was never caught before.